echopype/tests/test_preprocess.py

```diff
@@ -0,0 +1,178 @@
+    for f in range(nfreq):
+        data[f] = data[0] * (f + 1)
+
+    data_log = 10 * np.log10(data)      # Convert to log domain
+    freq_index = np.arange(nfreq)
+    ping_index = np.arange(npings)
+    range_bin = np.arange(nrange)
+    Sv = xr.DataArray(data_log, coords=[('frequency', freq_index),
+                                        ('ping_time', ping_index),
+                                        ('range_bin', range_bin)])
+    Sv.name = "Sv"
+    ds_Sv = Sv.to_dataset()
+    ds_Sv = ds_Sv.assign(
+        range=xr.DataArray(np.array([[np.linspace(0, 10, nrange)] * npings] * nfreq),
+                           coords=Sv.coords)
+    )
+
+    # Binned MVBS test
+    ds_MVBS = ep.preprocess.compute_MVBS_index_binning(
+        ds_Sv,
+        range_bin_interval=range_bin_num,
+        ping_num_interval=ping_num
+    )
+    data_test = (10 ** (ds_MVBS.Sv / 10))    # Convert to linear domain
+
+    # Shape test
+    shape = np.ceil((nfreq, npings / ping_num, nrange / range_bin_num)).astype(int)
+    assert np.all(data_test.shape == shape)
+
+    # Construct test array that increases by 1 for each range_bin and ping_time
+    test_array = np.repeat(np.add(*np.indices((shape[1], shape[2])))[None, ...] + 1, nfreq, axis=0)
+    # Increase test data by a multiple each frequency
+    for f in range(nfreq):
+        test_array[f] = (test_array[0] * (f + 1))
+
+    # Test all values in MVBS
+    np.allclose(data_test, test_array, rtol=0, atol=1e-12)
+
+
+def test_compute_MVBS():
+    """Test compute_MVBS on toy data"""
+
+    # Parameters for fake data
+    nfreq, npings, nrange = 4, 100, 4000
+    range_meter_bin = 7          # range in meters to average over
+    ping_time_bin = 3            # number of seconds to average over
+    ping_rate = 2                # Number of pings per second
+    range_bin_per_meter = 30     # Number of range_bins per meter
+
+    # Useful conversions
+    ping_num = npings / ping_rate / ping_time_bin                      # number of pings to average over
+    range_bin_num = nrange / range_bin_per_meter / range_meter_bin     # number of range_bins to average over
+    total_range = nrange / range_bin_per_meter                         # total range in meters
+
+    # Construct data with values that increase with range and time
+    # so that when compute_MVBS is performed, the result is a smaller array
+    # that increases by a constant for each meter_bin and time_bin
+    data = np.ones((nfreq, npings, nrange))
+    for p_i, ping in enumerate(range(0, npings, ping_rate * ping_time_bin)):
+        for r_i, rb in enumerate(range(0, nrange, range_bin_per_meter * range_meter_bin)):
+            data[0, ping:ping + ping_rate * ping_time_bin, rb:rb + range_bin_per_meter * range_meter_bin] += r_i + p_i
+    # First frequency increases by 1 each row and column, second increses by 2, third by 3, etc.
+    for f in range(nfreq):
+        data[f] = data[0] * (f + 1)
```

Comment:
Since you are reusing this code block for two functions, it'll be good to wrap this in a private function to make the overall logic clear